tests/conformance/test_conformance.py

```diff
@@ -356,94 +549,95 @@ def _delete_retry_test(host, id):
 ########################################################################################################################################
 
 
-@pytest.mark.parametrize("test_data", _CONFORMANCE_TESTS)
-def test_conformance_retry_strategy(test_data):
-    host = _API_ACCESS_ENDPOINT
-    if host == _DEFAULT_STORAGE_HOST:
+def pytest_generate_tests(metafunc):
+    for test_data in _CONFORMANCE_TESTS:
+        scenario_id = test_data["id"]
+        m = "s{}method".format(scenario_id)
+        c = "s{}case".format(scenario_id)
+        s = "s{}".format(scenario_id)
+        if s in metafunc.fixturenames:
+            metafunc.parametrize(s, [scenario_id])
+        if m in metafunc.fixturenames:
+            metafunc.parametrize(m, test_data["methods"])
+        if c in metafunc.fixturenames:
+            metafunc.parametrize(c, test_data["cases"])
+
+
+def test_retry_s1_always_idempotent(s1, s1method, s1case):
```

Comment:
This feels like a leak of the schema into the logic of this test file. What's your rationale for including this?